scripts/src/public/middleware.ts

```diff
@@ -3,15 +3,29 @@ import * as bearerToken from "express-bearer-token";
 import * as httpContext from "express-http-context";
 
 import { logRequest, reportMetrics, requestLogger } from "../middleware";
-import { BlockchainEndpointChanged } from "../errors";
+import { WrongBlockchainVersion } from "../errors";
+import { AuthenticatedRequest } from "./auth";
+import { Application } from "../models/applications";
+import { NoSuchApp } from "../errors";
 
 export { notFoundHandler, generalErrorHandler, statusHandler } from "../middleware";
 
-const deprecationError = function(req: express.Request, res: express.Response, next: express.NextFunction) {
-	const shouldRaiseError = req.header("x-simulate-deprecation-error");
-	if (shouldRaiseError) {
-		throw BlockchainEndpointChanged("simulated deprecation");
+export const clientMigrationCheck = async function(req: AuthenticatedRequest, res: express.Response, next: express.NextFunction) {
+	const CLIENT_BLOCKCHAIN_HEADER = "x-kin-blockchain-version";
+	const blockchainVersionHeader = req.header(CLIENT_BLOCKCHAIN_HEADER);
+
+	const app = await Application.get(req.context.user.appId);
+	if (!app) { // cached per instance
```

Comment:
redundant comment is redundant

Reply:
Also I think you might not always have `req.context` - i.e. this might not be an `AuthenticatedRequest`. You should check that first